src/components/molecules/PoolTransactions.tsx

```diff
@@ -52,52 +71,55 @@ async function getSymbol(
 
 async function getTitle(
   ocean: Ocean,
-  row: PoolTransaction,
-  locale: string,
-  oceanTokenAddress: string
+  row: TransactionHistoryPoolTransactions,
+  locale: string
 ) {
-  const addRemoveSymbol = await getSymbol(
-    ocean,
-    row.tokenIn || row.tokenOut,
-    oceanTokenAddress
-  )
+  let title = ''
 
-  const title =
-    row.type === 'join'
-      ? `Add ${formatPrice(row.tokenAmountIn, locale)} ${addRemoveSymbol}`
-      : row.type === 'exit'
-      ? `Remove ${formatPrice(row.tokenAmountOut, locale)} ${addRemoveSymbol}`
-      : `Swap ${formatPrice(row.tokenAmountIn, locale)} ${await getSymbol(
-          ocean,
-          row.tokenIn,
-          oceanTokenAddress
-        )} for ${formatPrice(row.tokenAmountOut, locale)} ${await getSymbol(
-          ocean,
-          row.tokenOut,
-          oceanTokenAddress
-        )}`
+  if (row.event === 'swap') {
+    const inToken = row.tokens.filter((x) => x.type === 'in')[0]
+    const inTokenSymbol = await getSymbol(ocean, inToken.tokenAddress)
+    const outToken = row.tokens.filter((x) => x.type === 'out')[0]
+    const outTokenSymbol = await getSymbol(ocean, outToken.tokenAddress)
+    title += `Swap ${formatPrice(
+      Math.abs(inToken.value).toString(),
+      locale
+    )}${inTokenSymbol} for ${formatPrice(
+      Math.abs(outToken.value).toString(),
+      locale
+    )}${outTokenSymbol}`
+  } else {
+    for (let i = 0; i < row.tokens.length; i++) {
+      const tokenSymbol = await getSymbol(ocean, row.tokens[i].tokenAddress)
+      if (i > 0) title += '\n'
+      title += `${row.event === 'join' ? 'Add' : 'Remove'} ${formatPrice(
```

Comment:
just checked, the mentioned transactions are of event type `setup` so that's another event to account for

Reply:
Ah, i see, i'll add `setup` too. In the older implementation setup wasn't considered so that's why i had no ideea it actually existed

Reply:
Should the text  look like `join` ? Meaning `Add 21 Ocean / Add 9 DEFPEL-72` or create a different message?

Reply:
That 2nd one seems more clear